processing/src/main/java/org/apache/druid/query/timeseries/TimeseriesQueryQueryToolChest.java

```diff
@@ -404,11 +407,16 @@ public boolean isCacheable(TimeseriesQuery query, boolean willMergeRunners)
   @Override
   public RowSignature resultArraySignature(TimeseriesQuery query)
   {
-    return RowSignature.builder()
-                       .addTimeColumn()
-                       .addAggregators(query.getAggregatorSpecs())
-                       .addPostAggregators(query.getPostAggregatorSpecs())
-                       .build();
+
+    RowSignature.Builder rowSignatureBuilder = RowSignature.builder();
+    if (query.getDimensionSpec() != null) {
+      rowSignatureBuilder.addDimensions(Collections.singletonList(query.getDimensionSpec()));
```

Comment:
I think you diagnosed the bug right but the fix is a bit sketchy. If the Timeseries query accepts a DimensionSpec but then only uses it in the array signature, the following problems occur:

1. The input field, extractionFn, and decoration logic of the DimensionSpec are ignored.
2. The type might not actually be correct here; it will use the type from the DimensionSpec, but that might not match the actual type of the field, because the query engine isn't enforcing it.
3. The array signature should also match the maps returned from normal map-based responses, but this won't.

I think the idea of a special parameter to the Timeseries query that makes the time column have a different name is a good idea, though. Maybe instead this would work:

- Add an undocumented timeseries context parameter like `timestampResultField` that _adds_ a new field containing the timestamp as a long, with the given name, to both the map and array responses.
- Modify the SQL layer to generate this context parameter for timeseries queries when there is a time floor dimension.

Reply:
Thank you for the review. I think your suggestion with using a special parameter sounds good to me. Initially, I added `dimensionSpec` just to be similar to groupBy and TopN but you are right that we do not need `dimensionSpec` and it is confusing with how timeseries is using it compare to groupBy and TopN. All we need is a special field/parameter to know when we are doing time floor dimension. 
Can you explain a little bit about your first point with `adds a new field containing the timestamp as a long, with the given name`? What do you mean by the given name and do you mean the field containing the floor(timestamp) not the actual timestamp? Thanks!

Reply:
> What do you mean by the given name and do you mean the field containing the floor(timestamp) not the actual timestamp?

By "the given name" I meant the `timestampResultField`. (I was thinking you would do something like `"timestampResultField": "d0"`.)

It should contain the floored timestamp, which at the point the query results are being returned, is the same as the timestamp in the response (it has already been floored).

Reply:
Done